documentation/Using-the-SDK/Advanced/Creating-Your-Own-DataSource.md

```diff
@@ -0,0 +1,216 @@
+# Abstract
+
+This document outlines a scenario where you want to implement a `DataSource`
+beyond the `FileDataSource` and `DirectoryDataSource` provided by the SDK.
+
+# Motivation
+
+For users that have data stored somewhere other than files or directories, it is
+desirable for the SDK to treat their `DataSource` as a first class `DataSource`.
+For example, suppose that you are writing many `CustomDataSource`s that get
+their data from a SQL database or from a web service. Rather than having a file
+with a URI in it, it would be nice to say that your `Custom DataSource` uses a 
+`SqlServerDataSource` or a `WebServiceDataSource`.
+
+# Requirements
+
+In order to have your new `DataSource` recognized by the SDK, you must do the
+following:
+
+- Create a new class inheriting from `DataSource`.
+    - You may also implment `IDataSource` directly.
+- Create a new Attribute inheriting from `DataSourceAttribute`, and specify the
+  type of the class you created in the call to the base constructor.
```

Comment:
instead of saying "the type of the class you created" maybe say "the `DataSource` you created"